demo/core_32_$.jssp.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Code City: Templating system for Code City.
+ *
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview JavaScript Server Pages for Code City.
+ * @author frase@google.com (Neil Fraser)
+ */
+
+$.jssp = {};
+
+$.jssp.compile = function(func) {
+  var src = func.jssp;
+  if (!src) {
+    throw ReferenceError('jssp not found on function.');
+  }
+  var tokens = src.trim().split(/(<%(?:--|=)?|(?:--)?%>)/);
+  var code = ["// DON'T EDIT THIS CODE: AUTOMATICALLY GENERATED BY JSSP."];
+  var STATES = {
+    LITERAL: 0,
+    STATEMENT: 1,
+    EXPRESSION: 2,
+    COMMENT: 3
+  };
+  var state = STATES.LITERAL;
+  var buffer = '';
+  for (var i = 0; i < tokens.length; i++) {
+    var token = tokens[i];
+    if (!token) {
+      continue;  // Empty string caused by splitting adjacent tags.
+    }
+    switch (state) {
+      case STATES.LITERAL:
+        var flush = true;
+        if (token === '<%') {
+          state = STATES.STATEMENT;
+        } else if (token === '<%=') {
+          state = STATES.EXPRESSION;
+        } else if (token === '<%--') {
+          state = STATES.COMMENT;
+        } else {
+          buffer += token;
+          flush = false;
+        }
+        if (flush && buffer) {
```

Comment:
Is this worth the complexity?  It seems to me that you will only ever get one literal token in a row, so all the buffering etc. seems like a waste of time.  (And even if I'm wrong, in the worst case you'd just get extra calls to out.print(), which is not ideal for performance but hardly a disaster.)